projecttacoma/measure-repository
# Render JSON content for selected resource

# Summary
Adds dynamic route for rendering the JSON content of a selected resource on the server.

## New behavior
We now have a new page that has tabs for the JSON, ELM, CQL, and narrative content of a selected resource (all are disabled except for JSON for now - this will change once the other tab functionality is implemented). The JSON tab renders the content of the selected resource.

## Code changes
* New component `BackButton.tsx` to go back to the previous page (will bring the user back to the list of resources once that is merged)
* New page [id].tsx that renders the JSON content of the selected resource. Uses `getServerSideProps` to fetch the JSON content and stringify it. Uses a Mantine scrollable `Prism` to display the JSON content

# Testing guidance
* Create a local `.env.local` file in the frontend directory with the `NEXT_PUBLIC_MRS_SERVER` environment variable. You can use `.env.example` as an example
* Run `npm run start:all` to start the server and frontend (backend defaults to 3000, frontend defaults to 3001)
* Navigate to the url for the frontend app to access the application.
* **Note**: The resource list page from https://github.com/projecttacoma/measure-repository/pull/26 is not merged into main. To access the rendered JSON page, type the url into the browser box directly. The url should be of the form `<localhost url>/<resourceType>/<id>`. For example, if the `ColorectalCancerScreeningsFHIR` measure is on the server, navigate to `http://localhost:3001/Measure/ColorectalCancerScreeningsFHIR` and Nextjs will resolve the route.
* Ensure the JSON content properly loads.

For database loading (with an example measure bundle):
```
npm run db:reset
npm run db:loadBundle "ecqm-content-r4-2021/bundles/measure/ColorectalCancerScreeningsFHIR/ColorectalCancerScreeningsFHIR-bundle.json"
```
